pop_tools/xgcm_util.py

```diff
@@ -0,0 +1,169 @@
+"""
+Module for modifying pop model output to be compatible with xgcm
+
+# Ref: https://gist.github.com/rabernat/933bc785c99828352f343e48d0da6e22
+"""
+
+import numpy as np
+import xarray as xr
+
+
+def _add_pop_dims_to_dataset(ds):
+    ds_new = ds.copy()
+    ds_new['nlon_u'] = xr.Variable(
+        ('nlon_u'), np.arange(len(ds.nlon)) + 1, {'axis': 'X', 'c_grid_axis_shift': 0.5}
+    )
+    ds_new['nlat_u'] = xr.Variable(
+        ('nlat_u'), np.arange(len(ds.nlat)) + 1, {'axis': 'Y', 'c_grid_axis_shift': 0.5}
+    )
+    ds_new['nlon_t'] = xr.Variable(('nlon_t'), np.arange(len(ds.nlon)) + 0.5, {'axis': 'X'})
+    ds_new['nlat_t'] = xr.Variable(('nlat_t'), np.arange(len(ds.nlat)) + 0.5, {'axis': 'Y'})
+
+    # add metadata to z grid
+    if 'z_t' in ds_new.variables:
+        ds_new['z_t'].attrs.update({'axis': 'Z'})
+    if 'z_w' in ds_new.variables:
+        ds_new['z_w'].attrs.update({'axis': 'Z', 'c_grid_axis_shift': -0.5})
+    if 'z_w_top' in ds_new.variables:
+        ds_new['z_w_top'].attrs.update({'axis': 'Z', 'c_grid_axis_shift': -0.5})
+    if 'z_w_bot' in ds_new.variables:
+        ds_new['z_w_bot'].attrs.update({'axis': 'Z', 'c_grid_axis_shift': 0.5})
+
+    return ds_new
+
+
+def _dims_from_grid_loc(grid_loc):
+    grid_loc = str(grid_loc)
+    ndim = int(grid_loc[0])
+    x_loc_key = int(grid_loc[1])
+    y_loc_key = int(grid_loc[2])
+    z_loc_key = int(grid_loc[3])
+
+    x_loc = {1: 'nlon_t', 2: 'nlon_u', 3: 'nlon_u'}[x_loc_key]
+    y_loc = {1: 'nlat_t', 2: 'nlat_u', 3: 'nlat_t'}[y_loc_key]
+    y_loc = {1: 'nlat_t', 2: 'nlat_u'}[y_loc_key]
+    z_loc = {0: 'surface', 1: 'z_t', 2: 'z_w', 3: 'z_w_bot', 4: 'z_t_150m'}[z_loc_key]
+
+    if ndim == 3:
+        return z_loc, y_loc, x_loc
+    elif ndim == 2:
+        return y_loc, x_loc
+
+
+def _label_coord_grid_locs(ds):
+    grid_locs = {
```

Comment:
we need a dataset that tests all of these. I've emailed Anna on our previous email thread. She was looking at one that had DZU & DZT